server/ctrlsubsonic/ctrl.go

```diff
@@ -218,43 +220,159 @@ func withRequiredParams(next http.Handler) http.Handler {
 	})
 }
 
-func withUser(dbc *db.DB) handlerutil.Middleware {
-	return func(next http.Handler) http.Handler {
-		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			params := r.Context().Value(CtxParams).(params.Params)
-			// ignoring errors here, a middleware has already ensured they exist
-			username, _ := params.Get("u")
-			password, _ := params.Get("p")
-			token, _ := params.Get("t")
-			salt, _ := params.Get("s")
-
-			passwordAuth := token == "" && salt == ""
-			tokenAuth := password == ""
-			if tokenAuth == passwordAuth {
-				_ = writeResp(w, r, spec.NewError(10,
-					"please provide `t` and `s`, or just `p`"))
-				return
-			}
-			user := dbc.GetUserByName(username)
-			if user == nil {
+func (c *Controller) WithUser(next http.Handler) http.Handler {
```

Comment:
looks like this function signature has changed, it used to be `unc withUser(dbc *db.DB) handlerutil.Middleware {` and still can be by the looks of things